tests/components/modbus/test_modbus_sensor.py

```diff
@@ -650,3 +651,104 @@ async def test_service_sensor_update(hass, mock_pymodbus):
         "homeassistant", "update_entity", {"entity_id": entity_id}, blocking=True
     )
     assert hass.states.get(entity_id).state == "32"
+
```

Comment:
please rewrite this to do the from from “outside”. What you test looks good.

You can see in other tests how we test structures.

Reply:
Yeah, I will encapsulate the `create_data_structure` method inside the `ModbusRegisterSensor` to reduce the surface. Just want to figure out conceptual thing here https://github.com/home-assistant/core/pull/50234#discussion_r628850179

Reply:
👍